hibernate-core/src/main/java/org/hibernate/graph/EntityGraphs.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.graph;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.persistence.AttributeNode;
+import javax.persistence.EntityGraph;
+import javax.persistence.EntityManager;
+import javax.persistence.Subgraph;
+
+import org.hibernate.Session;
+import org.hibernate.engine.spi.SessionImplementor;
+import org.hibernate.graph.spi.GraphImplementor;
+import org.hibernate.graph.spi.RootGraphImplementor;
+
+/**
+ * A collection of {@link EntityGraph} utilities.
+ * These methods really belong inside other classes that we cannot modify, hence here.
+ * 
+ * @author asusnjar
+ */
+@SuppressWarnings("WeakerAccess")
+public final class EntityGraphs {
+
+	/**
+	 * The JPA-defined HINT key for specifying an EntityGraph to use applying
+	 * fetch semantics
+	 */
+	public static volatile String HINT_FETCHGRAPH = EntityGraphSemantic.FETCH.getJpaHintName();
+
+	/**
+	 * The JPA-defined HINT key for specifying an EntityGraph to use applying
+	 * load semantics
+	 */
+	public static volatile String HINT_LOADGRAPH = EntityGraphSemantic.FETCH.getJpaHintName();
```

Comment:
Should the two elements above be `final` instead of `volatile`?

Reply:
Perhaps.  Not to mention HINT_LOADGRAPH probably ought to use EntityGraphSemantic.LOAD :D.  Lot of copy/paste errors still.  This is just an initial preview ;)